src/main/java/hudson/plugins/jira/JiraSite.java

```diff
@@ -510,9 +500,32 @@ public String getName() {
      * @return null if remote access is not supported.
      */
     @Nullable
-    public JiraSession getSession() {
+    public JiraSession getSession(ItemGroup context) {
```

Comment:
I think we need to keep binary compatibility here, i.e. mark the public constructor as Deprecated and introduce the new one with context.

Reply:
I don't find the way to get the context in a DataBoundConstructor... Do you have any idea?

Reply:
if you have a `doFill` method you can use `@AncestorInPath ItemGroup context`
https://github.com/jenkinsci/hashicorp-vault-plugin/blob/8e70b761172b9b0cf72719aa9191d71d5fc01fbf/src/main/java/com/datapipe/jenkins/vault/configuration/VaultConfiguration.java#L150

Reply:
> 
> 
> if you have a `doFill` method you can use `@AncestorInPath ItemGroup context`
> https://github.com/jenkinsci/hashicorp-vault-plugin/blob/8e70b761172b9b0cf72719aa9191d71d5fc01fbf/src/main/java/com/datapipe/jenkins/vault/configuration/VaultConfiguration.java#L150

I need it in constructor if possible.

Reply:
What he means you need to keep `getSession()` 

I think for simplicity  `getSession()` should call `getSession(ItemGroup context)` with null.

so 
```java
public JiraSession getSession() {
    return getSession(null);
}
```